src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "ray/common/task/task.h"
+#include "ray/common/task/task_common.h"
+#include "ray/raylet/scheduling/cluster_resource_scheduler.h"
+#include "ray/raylet/worker.h"
+#include "ray/raylet/worker_pool.h"
+
+namespace ray {
+namespace raylet {
+
+typedef std::function<void(std::shared_ptr<Worker>, ClientID spillback_to,
+                           std::string address, int port)>
+    ScheduleFn;
+
+typedef std::pair<ScheduleFn, Task> Work;
+
+class ClusterTaskManager {
+ public:
+  /// fullfills_dependencies_func Should return if all dependencies are
+  /// fulfilled and unsubscribe from dependencies only if they're
```

Comment:
Maybe update the comment to say what the semantics are if the dependencies are not yet fulfilled.